CrossPlatform/BasePage.cs

```diff
@@ -14,7 +14,9 @@ public abstract class BasePage
         protected bool OnAndroid { get; private set; }
         protected bool OniOS { get; private set; }
 
-        protected Func<AppQuery, AppQuery> Trait { get; set; }
+        protected abstract Trait Trait { get; }
+
+        protected Func<AppQuery, AppQuery> CurrentTrait { get; private set; }
```

Comment:
I'd prefer not to have this secondary trait. We should just use `Trait` directly everywhere imo